packages/kit/src/core/sync/utils.js

```diff
@@ -25,9 +25,44 @@ export function write(file, code) {
 	fs.writeFileSync(file, code);
 }
 
-/** @param {string} str */
-export function trim(str) {
-	const indentation = /** @type {RegExpExecArray} */ (/\n?([ \t]*)/.exec(str))[1];
-	const pattern = new RegExp(`^${indentation}`, 'gm');
-	return str.replace(pattern, '').trim();
+/** @type {WeakMap<TemplateStringsArray, { strings: string[], indents: string[] }>} */
+const dedent_map = new WeakMap();
+
+/**
+ * @param {TemplateStringsArray} strings
```

Comment:
a comment here could help. is this the opposite of indent? why do we need it? won't Vite handle minimization for us? it seems like outputting minimized code in the intermediate step might make debugging harder

Reply:
added a comment. it's just a dedenting function. it takes a string like 

```js
			const code = dedent`
				const things = [
					${things.join('\n')}
				];
			`;
```

and makes it as though you wrote this

```js
			const code = `const things = [\n\t${things.join('\n\t')}\n];`
```